project/iterative_sorting.py

```diff
@@ -1,36 +1,54 @@
 # Complete the selection_sort() function below in class with your instructor
+sort_array = [5, 7, 8, 4, 10, 7, 11, 18, 2019, 48]
+sort_array2 = [50, 70, 80, 40, 100, 70, 110, 180, 20190, 480]
+
 def selection_sort( arr ):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
         # TO-DO: find next smallest element
         # (hint, can do in 3 loc) 
-             
-
-
-
+        for j in range(cur_index, len(arr)):
+            if arr[j] < arr[smallest_index]:
+                smallest_index = j
         # TO-DO: swap
-
-
-
+        temp = arr[smallest_index]
+        arr[smallest_index] = arr[cur_index]
+        arr[cur_index] = temp
 
     return arr
-
+print(selection_sort(sort_array))
 
 # TO-DO: implement the Insertion Sort function below
 def insertion_sort( arr ):
+    for i in range(1, len(arr) - 1):
+        cur_value = arr[i]
+        position = i
+
+        while position > 0 and arr[position - 1] > cur_value:
+            arr[position] = arr[position - 1]
+            position = position - 1
+        arr[position] = cur_value
 
     return arr
```

Comment:
This sorts most of them. For some reason, when I copied the first test array into another variable (because these sorts are in-place, meaning they mutate the original arrays), this is the output I got: `[4, 5, 7, 7, 8, 10, 11, 18, 2019, 48]`. You're close, but something's off. I think you tricked yourself by trying to sort an already-sorted array.

Reply:
I fixed it, some increment math was off.